### ISSUE TITLE
howto rebuild docker images

### ISSUE THREAD

**Issue by terinjokes** [istio/issues#201]
Right now it's very difficult to rebuild Docker images, for example to test patches to Envoy in the sidecar and ingress services. Image builds are split across repositories, there's little documentation on building them, and they have a reliance on GCS to thread through build artifacts.

Can we write documentation and improve this situation?

**Comment by ldemailly** [comment]
agreed the doc can use more improvements - specially the recently changed dev workflows
should be fixed by https://github.com/istio/istio/pull/3475
(confirm that helps)

**Comment by terinjokes** [comment]
@ldemailly AFAICT that PR doesn't address rebuilding envoy_debug or working around [pulling tarballs from GCS](https://github.com/istio/istio/blob/17d6d49e7d007a3614161a8c4d26683050d39fa8/bin/init.sh#L68-L107) while building.

**Comment by ldemailly** [comment]
ic, I updated your issue title accordingly

can you try to follow the chain of commands... the actual proxy is built in
https://github.com/istio/proxy
?

**Comment by terinjokes** [comment]
I can rebuild envoy-debug from istio/proxy, I can rebuild the images in istio/istio. After hacking around the shell scripts, I can even rebuild with my own envoy-debug images.

I created this issue because none of this was documented, it took me the better part of a day to figure out. Because I had to modify scripts to actually build these images, I'm not 100% sure I've build things that are release-like.

**Comment by ldemailly** [comment]
doc / better scripts pr welcome :)
